examples/nlp/mlm_with_bert.py

```diff
@@ -0,0 +1,272 @@
+"""
+Title: Masked Language Modeling with BERT
+Author: [Ankur Singh](https://twitter.com/ankur310794)
+Date created: 2020/09/03
+Last modified: 2020/09/03
+Description: Implement a Masked Language Modeling with BERT and train on TPU
+"""
+"""
+## Introduction
+
+Masked language modeling is a fill-in-the-blank task, where a model uses the context words surrounding a [MASK] token to try to predict what the [MASK] word should be.
+"""
+
+"""
+## Setup
+
+Install HuggingFace transformers via pip install transformers (version >= 3.1.0).
+"""
+
+
+from dataclasses import dataclass
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+from tensorflow.keras.optimizers import Adam
+from transformers import TFAutoModelWithLMHead, AutoTokenizer
+from transformers import pipeline
+from pprint import pprint
+
+
+"""
+## Set-up Configuration
+"""
+
+
+@dataclass
+class Config:
+    MAX_LEN = 128
+    BATCH_SIZE = 16  # per TPU core
+    TOTAL_STEPS = 2000  # thats approx 4 epochs
+    EVALUATE_EVERY = 200
+    LR = 1e-5
+    PRETRAINED_MODEL = "bert-base-uncased"
+
+
+flags = Config()
+AUTO = tf.data.experimental.AUTOTUNE
+
+
+"""
+## Set-up TPU Runtime
+"""
+
+
+def connect_to_TPU():
+    """Detect hardware, return appropriate distribution strategy"""
+    try:
+        # TPU detection. No parameters necessary if TPU_NAME environment variable is
+        # set: this is always the case on Kaggle.
+        tpu = tf.distribute.cluster_resolver.TPUClusterResolver()
+        print("Running on TPU ", tpu.master())
+    except ValueError:
+        tpu = None
+
+    if tpu:
+        tf.config.experimental_connect_to_cluster(tpu)
+        tf.tpu.experimental.initialize_tpu_system(tpu)
+        strategy = tf.distribute.experimental.TPUStrategy(tpu)
+    else:
+        # Default distribution strategy in Tensorflow. Works on CPU and single GPU.
+        strategy = tf.distribute.get_strategy()
+
+    global_batch_size = flags.BATCH_SIZE * strategy.num_replicas_in_sync
+
+    return tpu, strategy, global_batch_size
+
+
+tpu, strategy, global_batch_size = connect_to_TPU()
+print("REPLICAS: ", strategy.num_replicas_in_sync)
+
+
+"""
+## Prepare Masked Language Dataset
+"""
+
+
+class Dataset:
```

Comment:
`Dataset` is already a base class in TF. I would recommend simply refactoring this as separate utility functions, since there is no need for an OO approach here.

Reply:
Ok, I have implemented it. Please review functions